components/Navbar/UserButton.tsx

```diff
@@ -50,8 +52,11 @@ export default function UserButton() {
   }
 
   return (
-    <div>
-      <div className="ml-4 h-10 w-10">{renderIconButton()}</div>
+    <div className="flex-l flex h-full items-stretch">
+      <span aria-describedby="navbar-tab-user">
+        <span>{renderIconButton()}</span>
```

Comment:
Lets try to stay away from these types of render functions, I feel they make it difficult to grasp the overall html structure at a glance for other contributors.

I would also strongly suggest to keep the markup separate from the component's logic. In this instance I would classify `toggleModal` and `hideModal` as component logic, whereas `renderIconButton` is markup so it should come after `toggleModal` and `hideModal`. Ideally we just inline it like this though:

```jsx
<span aria-describedby="navbar-tab-user">
  {!hasMounted && <div />}
  {hasMounted && isLoggedIn && (
    <button
      id="navbar-tab-user-icon"
      onClick={() => setOpenSignInModal(true)}
      aria-label="profile"
      className="text-grey-300 relative flex h-full w-12 cursor-pointer items-center justify-center md:w-16"
    >
      <Avatar avatar={user.avatar} size={30} />
    </button>
  )}
  {hasMounted && !isLoggedIn && (
    <button
      id="navbar-tab-user-login"
      onClick={() => toggleModal(true)}
      aria-label="profile"
      className="text-grey-300 relative flex h-full w-12 cursor-pointer items-center justify-center md:w-16"
    >
      <UserIcon />
    </button>
  )}
</span>
```